### PR TITLE
Scenci

### PR DESCRIPTION
**Pull request by gidden**
Updating our integration tests to include multiple test cases and a scenario with a carbon budget

### CODE REVIEW

**Comment by gidden** [comment]
Hi @danielhuppmann and/or @khaeru if you could please take a look, that would be great.

To note: **please do not squash and merge** these should be two separate commits. The first enables the ability to pass options to the cplex solver (yes, we will want to refactor this eventually to be solver agnostic). The second adds an additional test to our scenario suite utilizing the barrier method for solving. @danielhuppmann you should be able to build off of this to test the pricing changes.

Finally, i realize that I forgot to add release notes. Once this passes CI, I will do so, make the commits correct, and then we can merge it in.

**Comment by khaeru** [comment]
@gidden the documentation says ([here](https://message-ix.readthedocs.io/en/master/framework.html#running-a-model)) that `message_ix.Scenario.solve()` is only one of three recognized ways of running the model. Does deletion of the file cplex.opt muck with the other two?

**Comment by gidden** [comment]
> 
> 
> @gidden the documentation says ([here](https://message-ix.readthedocs.io/en/master/framework.html#running-a-model)) that `message_ix.Scenario.solve()` is only one of three recognized ways of running the model. Does deletion of the file cplex.opt muck with the other two?

Ah, good point. I suspect it will still run (with defaults), but perhaps it is better to keep it as is. To note, the first time a user solves with the `solve` function, that original cplex.opt will get overwritten..

**Comment by danielhuppmann** [comment]
> @gidden the documentation says ([here](https://message-ix.readthedocs.io/en/master/framework.html#running-a-model)) that `message_ix.Scenario.solve()` is only one of three recognized ways of running the model. Does deletion of the file cplex.opt muck with the other two?

@khaeru I would think that GAMS will write a warning message to the log file but still solve

File: message_ix/core.py [hunk 1]
```diff
@@ -272,16 +283,38 @@ def valid(y_v, y_a):
...
+        # https://www.gams.com/latest/docs/UG_GamsCall.html#GAMSAOoptfile);
+        # however, we need to clean up the code in ixmp that passes arguments
+        # to gams to do so.
+        fname = os.path.join(default_paths.model_path(), 'cplex.opt')
+        opts = copy.deepcopy(DEFAULT_SOLVE_OPTIONS)
+        opts.update(solve_options)
+        lines = '\n'.join('{} = {}'.format(k, v) for k, v in opts.items())
+        with open(fname, 'w') as f:
+            f.writelines(lines)
         return super(Scenario, self).solve(model=model, **kwargs)
```
[the remark this answers is not in the document]
Comment (gidden) [reply message_ix/core.py]: done!

**Comment by gidden** [comment]
Ok, this should now be good to go (after being reviewed)

**Comment by gidden** [comment]
@danielhuppmann when this passes CI, I will "rebase and merge" so if you want to rebase #172 on this branch, you should be safe to do so

**Comment by gidden** [comment]
ok @danielhuppmann all is good to go here

**Pull request merged**